src/ToDo.UI/Views/SettingsPage.xaml.cs

```diff
@@ -1,9 +1,33 @@
 ﻿namespace ToDo.Views;
+using Uno.Toolkit.UI;
 
 public sealed partial class SettingsPage : Page
 {
 	public SettingsPage()
 	{
 		this.InitializeComponent();
+
+		this.Loaded += (s, e) =>
+		{
+			// Set default theme
+			var currentTheme = SystemThemeHelper.IsRootInDarkMode(XamlRoot) ? "Dark" : "Light";
```

Comment:
set a flag to prevent calling the update-xyz functions during this initialization

Reply:
alternatively move the ItemChecked handler registration here, after you have done the initialization part